rkt/api_service.go

```diff
@@ -0,0 +1,557 @@
+	cmdAPIService.Flags().StringVar(&flagAPIServiceListenClientURL, "--listen-client-url", common.APIServiceListenClientURL, "address to listen on client API requests")
+}
+
+// v1AlphaAPIServer implements v1Alpha.APIServer interface.
+type v1AlphaAPIServer struct {
+	store *store.Store
+}
+
+var _ v1alpha.PublicAPIServer = &v1AlphaAPIServer{}
+
+func newV1AlphaAPIServer() (*v1AlphaAPIServer, error) {
+	s, err := store.NewStore(globalFlags.Dir)
+	if err != nil {
+		return nil, err
+	}
+
+	return &v1AlphaAPIServer{
+		store: s,
+	}, nil
+}
+
+// GetInfo returns the information about the rkt, appc, api server version.
+func (s *v1AlphaAPIServer) GetInfo(context.Context, *v1alpha.GetInfoRequest) (*v1alpha.GetInfoResponse, error) {
+	return &v1alpha.GetInfoResponse{
+		Info: &v1alpha.Info{
+			RktVersion:  version.Version,
+			AppcVersion: schema.AppContainerVersion.String(),
+			ApiVersion:  supportedAPIVersion,
+		},
+	}, nil
+}
+
+type valueGetter interface {
+	Get(string) (string, bool)
+}
+
+// findKeyValue returns true if the actualKVs contains any of the key-value
+// pairs listed in requiredKVs, otherwise it returns false.
+func findKeyValue(actualKVs valueGetter, requiredKVs []*v1alpha.KeyValue) bool {
+	for _, requiredKV := range requiredKVs {
+		actualValue, ok := actualKVs.Get(requiredKV.Key)
+		if ok && actualValue == requiredKV.Value {
+			return true
+		}
+	}
+	return false
+}
+
+// findString tries to find a string in a string array which satisfies the checkFunc.
+// The checkFunc takes two strings, and returns whether the two strings satisfy the
+// given condition.
+func findString(needle string, haystack []string, checkFunc func(a, b string) bool) bool {
+	for _, v := range haystack {
+		if checkFunc(needle, v) {
+			return true
+		}
+	}
+	return false
+}
+
+// stringEqual returns true if two strings are equal.
+func stringEqual(a, b string) bool {
+	return a == b
+}
+
+// hasBaseName returns true if the second string is the base name of
+// the first string.
+func hasBaseName(name, baseName string) bool {
+	return path.Base(name) == baseName
+}
+
+// findAnyString returns true if there's any two-string pair from array a and
+// array b that satisfy the checkFunc.
+//
+// e.g. if a = {"a", "b", "c"}, b = {"c", "d", "e"} and c = {"e", "f", "g"},
+// then findAnyString(a, b, stringEqual) == true,
+//      findAnyString(a, c, stringEqual) == false,
+//      findAnystring(b, c, stringEqual) == true.
+//
+func findAnyString(a, b []string, checkFunc func(a, b string) bool) bool {
+	for _, aa := range a {
+		if findString(aa, b, checkFunc) {
+			return true
+		}
+	}
+	return false
+}
+
+// filterPod returns true if the pod doesn't satisfy the filter, which means
+// it should be filtered and not be returned.
+// It returns false if the filter is nil or the pod satisfies the filter, which
+// means it should be returned.
+func filterPod(pod *v1alpha.Pod, manifest *schema.PodManifest, filter *v1alpha.PodFilter) bool {
```

Comment:
Would you mind splitting this function into shorter chunks? Like `filterPod` would call `filterPodState`, `filterPodAppNames` and so on?